### PR TITLE
Deflake Shoot Maintenance controller integration test

### LINKED ISSUE #6659: [Flaky Test] Shoot Maintenance controller tests
**How to categorize this issue?**

/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:
```
  Worker Pool Kubernetes version maintenance tests
  /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:417
    [It] Kubernetes version should be updated: force update patch version
    /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:442
...
 Expected
      <string>: 0.0.1
  to equal
      <string>: 0.0.5
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:454
```
and 
```
  Machine image maintenance tests
  /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:322
    [It] Shoot machine image must be updated in maintenance time: AutoUpdate.MachineImageVersion == false && expirationDate applies
    /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:348
...
  Expected success, but got an error:
      <*errors.errorString | 0xc0014f1810>: {
          s: "Assertion in callback at /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:356 failed:\nExpected\n    <v1beta1.ShootMachineImage>: {\n        Name: \"foo-image\",\n        ProviderConfig: nil,\n        Version: \"0.0.1-beta\",\n    }\nto equal\n    <v1beta1.ShootMachineImage>: {Name: \"foo-image\", ProviderConfig: nil, Version: \"1.1.1\"}",
      }
      Assertion in callback at /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:356 failed:
      Expected
          <v1beta1.ShootMachineImage>: {
              Name: "foo-image",
              ProviderConfig: nil,
              Version: "0.0.1-beta",
          }
      to equal
          <v1beta1.ShootMachineImage>: {Name: "foo-image", ProviderConfig: nil, Version: "1.1.1"}
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/integration/controllermanager/shoot/maintenance/maintenance_test.go:358
```
**CI link**:
https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/6638/pull-gardener-integration/1568990366986670080

**Reason for failure**:
Mentioned above.

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area testing
/kind flake

**What this PR does / why we need it**:
See #6659

**Which issue(s) this PR fixes**:
Fixes #6659

On the yesterdays HEAD (203027d0526b2bf3ebcf1a08d6ac69e9c15a0a66):
```
$ stress -p 32 ./test/integration/controllermanager/shoot/maintenance/maintenance.test -ginkgo.progress
...
10m20s: 422 runs so far, 19 failures (4.50%)
```

With this PR:
```
$ stress -p 32 ./test/integration/controllermanager/shoot/maintenance/maintenance.test -ginkgo.progress
...
21m20s: 873 runs so far, 0 failures
```

**Special notes for your reviewer**:
The PR also includes 2 cleanup small commits that are not directly related to the deflaking but I decided to add them as this PR as it does not make sense to create a new PR for such small changes.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: test/integration/controllermanager/shoot/maintenance/utils_test.go [hunk 1]
```diff
@@ -40,6 +41,33 @@ func waitForShootToBeMaintained(shoot *gardencorev1beta1.Shoot) {
 	}).Should(BeFalse())
 }
 
+func waitMachineImageVersionToBeExpiredInCloudProfile(cloudProfileName, imageName, imageVersion string, expirationDate *metav1.Time) {
+	Eventually(func(g Gomega) {
```
Comment (rfranzke) [inline test/integration/controllermanager/shoot/maintenance/utils_test.go]: Should this be `EventuallyWithOffset` as described in https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-tests-relevant-for-all-kinds:

> use `ExpectWithOffset` for making assertions in helper funcs like `expectSomethingWasCreated`: [example test](https://github.com/gardener/gardener/blob/2eb54485231408cbdbabaa49812572a07124364f/extensions/pkg/controller/controlplane/genericactuator/actuator_test.go#L732-L736)

(same below)

**Review by rfranzke — COMMENTED** [review]
Overall this looks good, also thanks for cleaning up the nits :)

File: test/integration/controllermanager/shoot/maintenance/utils_test.go (hunk 1 above)
Comment (ialidzhikov) [reply test/integration/controllermanager/shoot/maintenance/utils_test.go]: Good point, thank you!

**Pull request merged**